extensions/iceberg/src/main/java/io/deephaven/iceberg/util/BuildCatalogOptions.java

```diff
@@ -30,11 +31,39 @@ public String name() {
     }
 
     /**
-     * The catalog properties. Must contain {@value CatalogUtil#ICEBERG_CATALOG_TYPE} or
+     * The catalog properties provided by the user. Must contain {@value CatalogUtil#ICEBERG_CATALOG_TYPE} or
      * {@value CatalogProperties#CATALOG_IMPL}.
+     * <p>
+     * For building the catalog, use {@link #updatedProperties()} instead.
      */
     public abstract Map<String, String> properties();
 
+    abstract BuildCatalogOptions withProperties(Map<String, ? extends String> entries);
+
+    /**
+     * Enables Deephaven’s automatic injection of properties that work around upstream issues and supply defaults needed
+     * for Deephaven’s Iceberg usage. Disable to manage all properties yourself.
+     */
+    @Value.Default
+    public boolean enablePropertyInjection() {
+        return true;
+    }
+
+    /**
+     * The properties to use when creating the catalog. If {@link #enablePropertyInjection} is {@code true}, this
+     * creates a copy of the {@link #properties()} map, with any Deephaven-specific properties injected into it. Else,
+     * returns the original {@link #properties()} map.
+     */
+    public final Map<String, String> updatedProperties() {
+        if (!enablePropertyInjection()) {
+            return properties();
+        }
+        final Map<String, String> updatedProperties = new LinkedHashMap<>(properties());
+        // Inject Deephaven-specific AWS/S3 settings into the property map
+        AWSProperties.injectDeephavenProperties(updatedProperties);
+        return updatedProperties;
+    }
```

Comment:
Thinking about the best structuring for this, I think it's best to remove this method from the options and create / document `static Map<String, String> injectDeephavenProperties(Map<String,String>)` in IcebergTools (and retarget where the testing is done).